tests/integration/test_charm.py

```diff
@@ -2,20 +2,49 @@
 # See LICENSE file for licensing details.
 
 import logging
+from pathlib import Path
 
 import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
 
 log = logging.getLogger(__name__)
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+DEX_CONFIG = {
+    "static-username": "admin",
+    "static-password": "foobar",
+}
+OIDC_CONFIG = {
+    "client-name": "Ambassador Auth OIDC",
+    "client-secret": "oidc-client-secret",
+}
 
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test):
     my_charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy(my_charm, "dex-auth-operator", trust=True)
-    await ops_test.model.wait_for_idle()
+    await ops_test.model.deploy(my_charm, trust=True, config=DEX_CONFIG)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=600
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
 
-async def test_status(ops_test):
-    for app in ops_test.model.applications.values():
-        for unit in app.units:
-            assert unit.workload_status == "active"
+@pytest.mark.abort_on_fail
+async def test_relations(ops_test: OpsTest):
```

Comment:
I would like to expand this test a bit more, it should contain all the relations between dex/oidc  and isito (the ones between istio and oidc are missing), and some check that we can actually access the dex dashboard.

Also @ca-scribner , can you help @agathanatasha to setup these tests so that we could reuse them in the full bundle in the future?

Reply:
will address this in [this ticket](https://warthogs.atlassian.net/browse/KF-198) as discussed in planning